Release 1.0.0 (#7)
vcmi/VCMI.eu: content/news/vcmi-0.100-release.md

+++
categories = ["Releases", "Stable"]
date = "2022-09-10"
description = ""
featured = "vcminewterrain.jpg"
featuredalt = ""
featuredpath = "date"
linktitle = ""
title = "VCMI 0.100 released"
type = "post"
forumurl = "https://forum.vcmi.eu/"

+++

## We're happy to announce new release!

Few years passed after we released VCMI 0.99 and now it's time for a new release.

> Tons of changes, improvements, and fixes have been introduces during that time, it's even hard to recall all of them.

Reviewer: ```suggestion
Tons of changes, improvements, and fixes have been introduced during that time, it's even hard to recall all of them.
```

However we'd like to highlight several major changes.

> 

Reviewer: ## Game engine

Lodestar grail finally can reveal all the map!
We were also working hard to track down and fix the most of the crashes constantly hurting VCMI gameplay. Now they are rare beasts but if you manage to find one do not hesitate to report it!

Reviewer: I also think if we should mention our new buttons on hero exchange window. Something like
## Adventure map

Hero exchange window has got a few additional buttons and keyboard shortcuts to ease army and artifact exchange. If you do not see them after installing the release ensure you have latest VCMI essentials mod.

## Artificial intellengence

> You are welcome to try brand new AI algorithm called Nullkiller or you can play with classic VCAI computer player.

Reviewer: ```suggestion
You are welcome to try an alternative AI algorithm called Nullkiller which shows itself well on small to medium maps. For larger maps you should better go with classic VCAI computer player which also became much smarter ...
```

Reviewer 2: `shows itself well on small to medium maps` in the sentence `to` seems strange. If there is no sizes between small and medium, I would write `shows itself well on small and medium maps`. If there are sizes between them, I would write something like `shows itself well on maps up to medium size`.

Author: Maybe "or"

Reviewer: shows itself well on small and medium maps - this sounds better, thank you

> Both algorithms became much smarter in comparison with last release, so just choose from the laucher settings whatever you like and enjoy.

Reviewer: ```suggestion
Both algorithms became much smarter in comparison with the last release, so just choose from the launcher settings whatever you like and enjoy.
```

Reviewer 2: Something does not click: first, you write "NullKiller is a **brand** new algorithm", and now you write "Both algorithms got smarter in comparison to the last release". So, the question is "is NullKiller a new algorithm that appeared in 0.100? or it was included into 0.99?". In the former case, "Both got smarter..." is false, in the latter case "a brand new" is not true.

Reviewer: became much smarter -> was significantly improved

## Mods and new terrains

> VCMI opens broad opportunitites for modding. Since last release we significantly empowered mod system.

Reviewer: ```suggestion
VCMI opens broad opportunitites for modding. Since the last release we significantly empowered the mod system.
```

> Among various gameplay tools we introduced support of new terrains. It is a natural evolution of new towns as they need to have unique terrains, battlefields, mines and map objects. This feature required deep refactoring of game engine so we all saves you made before release 0.100 will not be compatible. So it's a good reason to make new ones!

Reviewer: ```suggestion
Among various gameplay tools we introduced support of new terrains. It is a natural evolution of new towns as they need to have unique terrains, battlefields, mines and map objects. This feature required deep refactoring of game engine so unfortunately all the saves you made before release 0.100 will not be compatible. So it's a good reason to make new ones!
```

## Random Maps Generator
RMG also has undergone huge changes. First of all, water modes are supported so all options from game menu work well.

> Of course supporting new terrains from mods is useless if you can't play with it. RMG has powerful algorithms to work with new terrains and placing new objects. Moreover, we improved generator in terms of aesthetics, obstacles placement and stability as well.

Reviewer 2: Nit: this sentence goes as if it continues some previous idea. But the previous sentence is about water maps and it is not related to new terrains. it confused me and it took some time to get the entire paragraph clearly. Maybe, rewrite it a bit. As an example (replace two first sentences from L31 with this): 

As mods can introduce new terrains and objects, RNG got powerful algorithms to generate new terrain patches and to place new objects provided by mods.

> ## TBD Alsphat mod???

Author: This block will not be here at the end, it's a placeholder I'll remove lated

TBD - I suggest to make "official" mod demonstrating new abilities.

> On my opinion, we need to make fine-quality mod with new town, new terrain, battlefields, mines, map objects (alphalt + forge for example). Also, we need to prepare some scenarios/maps (or event campaign) and some art (main menu?)

Reviewer 2: Looks like typo: `or event campaign`. I think it should be `or even campaign`
